homeassistant/components/envisalink/alarm_control_panel.py

```diff
@@ -113,31 +135,53 @@ def _update_callback(self, partition):
         if partition is None or int(partition) == self._partition_number:
             self.async_schedule_update_ha_state()
 
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        state_attr = {}
+
+        if self.state == STATE_ALARM_PENDING:
+            state_attr[ATTR_PRE_PENDING_STATE] = self._previous_state
+            state_attr[ATTR_POST_PENDING_STATE] = self._state
+            state_attr[CONF_PENDING_TIME] = self._pending_time
```

Comment:
Attributes can not contain counting values. Pending time is from a certain event. So you need to store a timestamp to when the pending time has expired.

Reply:
looks like other use of it is time_period, is that ok for attributes?

Reply:
you need to store the timestamp when pending time is finished. So `(utcnow() + pending_time).isoformat()`